12.js

```diff
@@ -0,0 +1,31 @@
+const stringMeAlong = string => {
+  let longestSubstring = ''
+  let substring
+
+  for (let index = 0; index < string.length; index++) {
```

Comment:
you seem to be looping `string.length * 2` times. I think you can do this without looping more that `string.length` times.